boards/arm/thingy91_nrf9160/Kconfig.board

```
@@ -40,6 +40,15 @@ config THINGY91_STATIC_PARTITIONS_LWM2M_CARRIER
 	help
 	  Use a partition layout including a storage partition needed for the lwm2m carrier library.
 
+config THINGY91_PARTITION_MANAGER
+	bool "Default partition manager logic [EXPERIMENTAL]"
```

Not sure default is the best wording here.

Maybe the word `dynamic` is better, as that is what is used in the PM guide: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/scripts/partition_manager/partition_manager.html

```suggestion
	bool "Dynamic partition manager logic [EXPERIMENTAL]"
```

---

I struggled how to best formulate this. Dynamic is not precise, because it's still possible to define static partitions from the application if they want to. The PM_STATIC_YML_FILE define we use in the CMakeLists.txt has the top level priority in partition manager logic - which makes it impossible to define an application static partition. There seems to be no way to make a board static partition which is possible to override from the application. The THINGY91_PARTITION_MANAGER config is used to so it's possible to not force one of the board static partitions.

---

I see what you mean, and upon second look I agree that `Dynamic` is not the best word either.

I still dislike the use `default` in this context.
As a user, it gives me the impression that I can go lookup somewhere to see what the default layout will be, which is not the case.
Here default doesn't refer to a given layout (as opposed to the choice options above).

When looking at this again, and the `choice` title:
https://github.com/nrfconnect/sdk-nrf/blob/154f3c80854f4416a0c8cef8e1d9307d9799c51b/boards/arm/thingy91_nrf9160/Kconfig.board#L18

How about changing that title to:
```
Fixed Thingy:91 partition layout
```
or 
```
Pre-defined Thingy:91 partition layout
```
Then the final choice could then be: `None`, and look as:
```
config THINGY91_PARTITION_MANAGER
	bool "None [EXPERIMENTAL]"
	select EXPERIMENTAL
	help
	   Don't use a pre-defined static partition layout, but use the default Partition Manager logic to
	   handle the partition layout. This allows the application to use dynamic layout or define
	   a custom static partition layout for the application. A debugger is needed to flash
	   Thingy:91 with a different partition layout.
```

And the menu, seen by the user will look:
```
Pre-defined Thingy:91 partition layout

  [X] "Factory Thingy:91 partition layout"
  [ ] "Secure boot Thingy:91 partition layout [EXPERIMENTAL]"
  [ ] "LWM2M Carrier partition layout"
  [ ] "None [EXPERIMENTAL]"
```

an alternative to `None` could be the word `auto`.

---

Thanks! That sounds like a good way to solve it. I also renamed the kconfig option itself from `THINGY91_PARTITION_MANAGER` to `THINGY91_NO_PREDEFINED_LAYOUT`.